File: devops/instrumentation/app/booking/internal/booking/booking.go
package booking

import "time"

type Booking struct {
	ID          string        `json:"id" gorm:"type:char(36);primaryKey"`
	MemberCode  string        `json:"member_code" gorm:"type:varchar(8);not null"`
	CourseCode  string        `json:"course_code" gorm:"type:varchar(8);not null"`
	BookingDate time.Time     `json:"booking_date" gorm:"type:date;not null"`
	Status      BookingStatus `json:"status" gorm:"type:int;not null;default:0"`
	Notes       string        `json:"notes" gorm:"type:text"`
	CreatedAt   time.Time     `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt   time.Time     `json:"updated_at" gorm:"autoUpdateTime"`
}

type BookingStatus int

const (
	BookingStatusPending BookingStatus = iota
	BookingStatusConfirmed
	BookingStatusCancelled
)

func (s BookingStatus) String() string {
	return [...]string{"pending", "confirmed", "cancelled"}[s]
}

File: __tmp__/devops/instrumentation/java-stack/app/booking/src/main/java/com/example/booking/domain/Booking.java
package com.example.booking.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "Bookings")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
public class Booking {

    @Id
    @Column(nullable = false, updatable = false)
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @Column(nullable = false)
    private String memberCode;

    @Column(nullable = false)
    private String courseCode;

    @Column(nullable = false)
    private LocalDateTime bookedAt;

    @Enumerated(EnumType.STRING)
    private Status status;

    private String notes;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private OffsetDateTime updatedAt;

    public enum Status {
        PENDING,
        CONFIRMED,
        CANCELLED
    }
}
